tests/integration_tests/validation/validation_util.py

```diff
@@ -0,0 +1,37 @@
+import os
+import pytest
+
+from arelle.CntlrCmdLine import parseAndRun
+from arelle import ModelDocument
+
+
+def get_test_data(args):
+    """
+    Produces a list of Pytest Params that can be fed into a parameterized pytest function
+
+    :param args: The args to be parsed by arelle in order to correctly produce the desired result set
+    :type args: list of strings
+    :return: A list of PyTest Params that can be used to run a parameterized pytest function
+    :rtype: list of ::class:: `~pytest.param`
+    """
+    cntlr = parseAndRun(args)
+    results = []
+    model_document = cntlr.modelManager.modelXbrl.modelDocument
```

Comment:
Can we use a `with` block with this, not entirely sure?

Reply:
I thought about this and checked to see if Cntlr has the necessary functions to use in a with block. It does not. This can be a future update.